### PR TITLE
Stop Building Components/E2E tests in the main pipeline

### LINKED ISSUE #38090: Stop Building Components/E2E tests in the main pipeline
The tests were disabled from the main/quarantine pipelines via https://github.com/dotnet/aspnetcore/pull/37891.

This issue tracks not building them either. 

Should help towards resolving https://github.com/dotnet/aspnetcore/issues/32219.

### PR DESCRIPTION
**Pull request by TanayParikh**
Fixes: https://github.com/dotnet/aspnetcore/issues/38090

### CODE REVIEW

**Comment by TanayParikh** [comment]
macOS playwright firefox build failure should be unrelated to this change (504 whilst downloading firefox). Left a note in https://github.com/dotnet/aspnetcore/issues/33527.

File: src/Components/test/E2ETestMigration/Microsoft.AspNetCore.Components.Migration.E2ETests.csproj [hunk 1]
```diff
@@ -12,7 +12,8 @@
     <TestTrimmedApps Condition="'$(ContinuousIntegrationBuild)' == 'true'">true</TestTrimmedApps>
     <TestTrimmedApps Condition="'$(Configuration)' == 'Release'">true</TestTrimmedApps>
 
-    <!-- Skip the Components E2ETestMigration in CI -->
+    <!-- Skip building and running the Components E2ETestMigration in CI -->
+    <ExcludeFromBuild Condition="'$(ContinuousIntegrationBuild)' == 'true'">true</ExcludeFromBuild>
```
Comment (pranavkm) [inline src/Components/test/E2ETestMigration/Microsoft.AspNetCore.Components.Migration.E2ETests.csproj:16]: Should we leave this building? I worry we wouldn't find out if this stops compiling if we turn off building on the CI. What would be preferable is to remove the `TestTrimmedApps` condition so we don't cause  trimming to happen as part of building this project. e.g.

```diff
- <TestTrimmedApps Condition="'$(ContinuousIntegrationBuild)' == 'true'">true</TestTrimmedApps>
- <TestTrimmedApps Condition="'$(Configuration)' == 'Release'">true</TestTrimmedApps>

- <ExcludeFromBuild Condition="'$(ContinuousIntegrationBuild)' == 'true'">true</ExcludeFromBuild>
```

File: src/Components/test/E2ETestMigration/Microsoft.AspNetCore.Components.Migration.E2ETests.csproj [hunk 2]
```diff
@@ -9,9 +9,6 @@
     <!-- This project references the shared framework transitively. Prevent restore errors by setting this flag. -->
     <GenerateErrorForMissingTargetingPacks>false</GenerateErrorForMissingTargetingPacks>
 
-    <TestTrimmedApps Condition="'$(ContinuousIntegrationBuild)' == 'true'">true</TestTrimmedApps>
-    <TestTrimmedApps Condition="'$(Configuration)' == 'Release'">true</TestTrimmedApps>
```
Comment (Pilchie) [inline src/Components/test/E2ETestMigration/Microsoft.AspNetCore.Components.Migration.E2ETests.csproj:13]: Does this mean we never test trimmed apps anymore?

Comment (TanayParikh) [reply src/Components/test/E2ETestMigration/Microsoft.AspNetCore.Components.Migration.E2ETests.csproj:13]: This is disabling it specifically for the `Migration` E2E tests (the ones written in playwright). These tests are a duplicate of our Selenium based tests, which still test the trimmed apps in the `components-e2e` pipeline:

https://github.com/dotnet/aspnetcore/blob/f8631573c0cecc2d4f732673af600aea3968935c/src/Components/test/E2ETest/Microsoft.AspNetCore.Components.E2ETests.csproj#L29-L30

Comment (Pilchie) [reply src/Components/test/E2ETestMigration/Microsoft.AspNetCore.Components.Migration.E2ETests.csproj:13]: Thanks!

**Review by pranavkm — APPROVED** [review]
🚢 🇮🇹

**Pull request merged**